src/Groups/matrices/form_group.jl

```diff
@@ -0,0 +1,629 @@
+         for i in 1:n, j in i+1:n
+            for h in 1:n
+               MM[pos_r,pos_c] = A[j,h]*B[i,h]+A[i,h]*B[j,h]
+               pos_c +=1
+            end
+            for h in 1:n, k in h+1:n
+               MM[pos_r,pos_c] = A[j,k]*B[i,h]+A[i,h]*B[j,k]+A[j,h]*B[i,k]+A[i,k]*B[j,h]
+               pos_c +=1
+            end
+            for h in 1:n, k in h+1:n
+               MM[pos_r,pos_c] = A[i,h]*A[j,k]+A[i,k]*A[j,h]+A[j,k]*B[i,h]+A[i,k]*B[j,h]+Nw*(B[i,h]*B[j,k]+B[i,k]*B[j,h])
+               pos_c +=1
+            end
+            pos_r +=1
+            pos_c =1
+         end
+ 
+         for i in 1:n^2
+            MM[i,i] -=1
+         end
+         push!(M,MM)
+      end
+   end
+
+   n_gens,K = nullspace(block_matrix(length(M),1,M))
+   N = div(n*(n-1),2)
+   L = T[]
+
+   for r in 1:n_gens
+      sol = zero_matrix(F,n,n)
+      for i in 1:n
+         sol[i,i] = e(K[i,r])
+      end
+      pos = n+1
+      if isodd(q)
+         for i in 1:n, j in i+1:n
+            sol[i,j] = e(K[pos,r])+w*e(K[pos+N,r])       # sol[i,j] = a_ij + w*b_ij
+            sol[j,i] = e(K[pos,r])-w*e(K[pos+N,r])
+            pos +=1
+         end
+      else
+         for i in 1:n, j in i+1:n
+            sol[i,j] = e(K[pos,r])+w*e(K[pos+N,r])       # sol[i,j] = a_ij + w*b_ij
+            sol[j,i] = e(K[pos,r])+e(K[pos+N,r])+w*e(K[pos+N,r])
+            pos +=1
+         end
+      end
+      push!(L,sol)
+   end
+
+   return L
+end
+
+# TODO 2nd approach: using MeatAxe GAP functionalities
+# TODO: these are not exported at the moment
+
+"""
+    function invariant_bilinear_form(G::MatrixGroup)
+
+Return an invariant bilinear form for the group `G`. It works only if the module induced by the action of `G` is absolutely irreducible.
+!!! warning "Note:"
+    At the moment, the output is returned of type `mat_elem_type(G)`.
+"""
+function invariant_bilinear_form(G::MatrixGroup)
+   V = GAP.Globals.GModuleByMats(GAP.Globals.GeneratorsOfGroup(G.X), G.mat_iso.fr.codomain)
+   B = GAP.Globals.MTX.InvariantBilinearForm(V)
+   return G.mat_iso(B)   
+end
+
+"""
+    function invariant_sesquilinear_form(G::MatrixGroup)
+
+Return an invariant sesquilinear (non bilinear) form for the group `G`. It works only if the module induced by the action of `G` is absolutely irreducible.
+!!! warning "Note:"
+    At the moment, the output is returned of type `mat_elem_type(G)`.
+"""
+function invariant_sesquilinear_form(G::MatrixGroup)
+   V = GAP.Globals.GModuleByMats(GAP.Globals.GeneratorsOfGroup(G.X), G.mat_iso.fr.codomain)
+   B = GAP.Globals.MTX.InvariantSesquilinearForm(V)
+   return G.mat_iso(B)   
+end
+
+"""
+    function invariant_quadratic_form(G::MatrixGroup)
+
+Return an invariant bilinear form for the group `G`. It works only if the module induced by the action of `G` is absolutely irreducible.
+!!! warning "Note:"
+    At the moment, the output is returned of type `mat_elem_type(G)`.
+"""
+function invariant_quadratic_form(G::MatrixGroup)
```

Comment:
Please add tests for this, and also take a look at the codecov comments visible on https://github.com/oscar-system/Oscar.jl/pull/368/files to discover more untested coode.